domain/snyk/delta/code_matcher.go

```diff
@@ -0,0 +1,277 @@
+	}
+
+	finalResult := deduplicateIssues(strongMatchingIssues)
+
+	// Assign identities found to results
+	for i, identity := range finalResult {
+		if !existingAssignedIds[identity.IdentityID] {
+			currentIssueList[i].SetGlobalIdentity(identity.IdentityID)
+		}
+	}
+
+	// Assign UUIDs for any new issues
+	return currentIssueList, nil
+}
+
+func findMatch(issue Identifiable, index int, baseIssueList []Identifiable, strongMatchingIssues map[string]IssueConfidence) {
+	matches := findMatches(issue, index, baseIssueList)
+
+	for _, match := range matches {
+		if existing, exists := strongMatchingIssues[match.BaseUUID]; !exists || existing.Confidence < match.Confidence {
+			strongMatchingIssues[match.BaseUUID] = match
+		}
+	}
+}
+
+func findMatches(currentIssue Identifiable, index int, baseIssues []Identifiable) IssueConfidenceList {
+	similarIssues := make(IssueConfidenceList, 0)
+
+	for _, baseIssue := range baseIssues {
+		if baseIssue.RuleId() != currentIssue.RuleId() {
+			continue
+		}
+
+		filePositionDistance := filePositionConfidence(baseIssue, currentIssue)
+		// Calculation of History is not needed here for IDE since we are not persisting old scan results.
+		//We will always return 1.
+		recentHistoryDistance := historicConfidenceCalculator()
+		fingerprintConfidence := fingerprintDistance(baseIssue, currentIssue)
+
+		overallConfidence := filePositionDistance*weights.FilePositionDistance +
+			recentHistoryDistance*weights.RecentHistoryDistance +
+			fingerprintConfidence*weights.FingerprintConfidence
+
+		if overallConfidence == 1 {
+			similarIssues = append(similarIssues, IssueConfidence{
+				BaseUUID:           baseIssue.GetGlobalIdentity(),
+				IssueIDResultIndex: index,
+				Confidence:         overallConfidence,
+			})
+			break
+		}
+
+		if overallConfidence > weights.MinimumAcceptableConfidence {
+			similarIssues = append(similarIssues, IssueConfidence{
+				BaseUUID:           baseIssue.GetGlobalIdentity(),
+				IssueIDResultIndex: index,
+				Confidence:         overallConfidence,
+			})
+		}
+	}
+
+	return similarIssues
+}
+
+func deduplicateIssues(strongMatchingIssues map[string]IssueConfidence) DeduplicatedIssuesToIDs {
+	finalResult := make(DeduplicatedIssuesToIDs)
+	for _, issue := range strongMatchingIssues {
+		if existing, exists := finalResult[issue.IssueIDResultIndex]; !exists || existing.Confidence < issue.Confidence {
+			finalResult[issue.IssueIDResultIndex] = Identity{
+				IdentityID: issue.BaseUUID,
+				Confidence: issue.Confidence,
+			}
+		}
+	}
+	return finalResult
+}
+
+func fingerprintDistance(baseFingerprints, currentFingerprints Identifiable) float64 {
+	baseFingerprintable, ok := baseFingerprints.(Fingerprintable)
+	if !ok {
+		return 0
+	}
+	currentFingerprintable, ok := currentFingerprints.(Fingerprintable)
+	if !ok {
+		return 0
+	}
+
+	// Split into parts and compare
+	parts1 := strings.Split(baseFingerprintable.GetFingerprint(), ".")
```

Comment:
Is `GetFingerprint` defined as in returning a fingerprint that contains `.` as separators?